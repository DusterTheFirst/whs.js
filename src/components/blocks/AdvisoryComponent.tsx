/*!
 * Copyright (C) 2018-2019  Zachary Kohnen (DusterTheFirst)
 */

import React from "react";
import { StyleSheet, Text, View } from "react-native";

const styles = StyleSheet.create({
    container: {
        backgroundColor: "#FFFFFF",
        // borderBottomColor: "#CECECE",
        // borderBottomWidth: 1,
        height: 75,
        padding: 10,
        width: "100%"
    },
    dim: {
        color: "#808080"
    },
    dualView: {
        alignItems: "baseline",
        flexDirection: "row",
        flexWrap: "nowrap",
        justifyContent: "space-between"
    },
    info: {
        marginTop: 5
    },
    room: {
        overflow: "visible"
    },
    teacher: {
        maxWidth: "50%"
    },
    title: {
        color: "#A0A0A0",
        fontSize: 25,
        fontWeight: "bold"
    }
});

export default function AdvisoryComponent({teacher, room}: IAdvisory) {
    return (
        <View style={styles.container}>
            <View style={styles.dualView}>
                <Text style={styles.title}>Advisory</Text>
                <Text style={styles.dim}>9:38 - 9:46 AM</Text>
            </View>
            <View style={[styles.dualView, styles.info]}>
                <Text style={[styles.dim, styles.teacher]} numberOfLines={1}>{teacher}</Text>
                <Text style={[styles.dim, styles.room]} numberOfLines={1}>Room {room}</Text>
            </View>
        </View>
    );
}