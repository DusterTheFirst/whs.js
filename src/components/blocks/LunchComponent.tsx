/*!
 * Copyright (C) 2018-2019  Zachary Kohnen (DusterTheFirst)
 */

import React, { memo } from "react";
import { Text, View } from "react-native";
import { classComponentStyles } from "../../styles/layout/default";
import { ITimes } from "../../util/class/extentions";

/** A component to display a lunch block */
function LunchComponent({ start, end }: ITimes) {
    return (
        <View style={classComponentStyles.container}>
            <View style={classComponentStyles.multiView}>
                <Text style={[classComponentStyles.title]}>Lunch</Text>
                <Text style={classComponentStyles.dim}>{start.format("h:mm")} - {end.format("h:mm A")}</Text>
            </View>
        </View>
    );
}

export default memo(LunchComponent);