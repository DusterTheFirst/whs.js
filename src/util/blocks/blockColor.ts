/*!
 * Copyright (C) 2018-2020  Zachary Kohnen (DusterTheFirst)
 */

/** Colors of the blocks */
export enum BlockColor {
    Red = "Red",
    Orange = "Orange",
    Yellow = "Yellow",
    Green = "Green",
    Blue = "Blue",
    Purple = "Purple",
    Tan = "Tan",
    None = "No Color"
}

/** A map of the block colors to their display colors */
type BlockColorColorMap = {
    readonly [B in BlockColor]: string;
};

/**
 * The display colors that corispond to each of the block colors
 */
const BlockColorDisplayColors: BlockColorColorMap = {
    [BlockColor.Blue]: "#2b5cbf",
    [BlockColor.Green]: "#5da539",
    [BlockColor.None]: "#999999",
    [BlockColor.Orange]: "#eb8a29",
    [BlockColor.Purple]: "#8848ad",
    [BlockColor.Red]: "#c93a3a",
    [BlockColor.Tan]: "#ad855e",
    [BlockColor.Yellow]: "#ddc923"
};

/** Lookup the display color for the given block color */
export function getDisplayColorForBlock(block: BlockColor | undefined) {
    return block === undefined ? undefined : BlockColorDisplayColors[block];
}