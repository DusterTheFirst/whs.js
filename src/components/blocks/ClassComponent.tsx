/*!
 * Copyright (C) 2018-2019  Zachary Kohnen (DusterTheFirst)
 */

import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { getDisplayColorForBlock } from "../../util/blocks/blockColor";
import { DisplayClass } from "../../util/class";

const styles = StyleSheet.create({
    container: {
        backgroundColor: "#FFFFFF",
        // borderBottomColor: "#CECECE",
        // borderBottomWidth: 1,
        height: 75,
        padding: 10,
        width: "100%"
    },
    dim: {
        color: "#808080"
    },
    dualView: {
        alignItems: "baseline",
        flexDirection: "row",
        flexWrap: "nowrap",
        justifyContent: "space-between"
    },
    info: {
        marginTop: 5
    },
    room: {
        overflow: "visible"
    },
    teacher: {
        maxWidth: "50%"
    },
    title: {
        color: "#A0A0A0",
        fontSize: 25,
        fontWeight: "bold"
    }
});

export default function ClassComponent({teacher, room, name, block, start, end}: DisplayClass) {
    return (
        <View style={styles.container}>
            <View style={styles.dualView}>
                <Text style={[styles.title, {color: getDisplayColorForBlock(block)}]}>{name}</Text>
                <Text style={styles.dim}>{start.format("h:mm")} - {end.format("h:mm A")}</Text>
            </View>
            <View style={[styles.dualView, styles.info]}>
                <Text style={[styles.dim, styles.teacher]} numberOfLines={1}>{teacher}</Text>
                <Text style={[styles.dim, styles.room]} numberOfLines={1}>Room {room}</Text>
            </View>
        </View>
    );
}