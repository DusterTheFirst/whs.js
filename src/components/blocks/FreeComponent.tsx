/*!
 * Copyright (C) 2018-2019  Zachary Kohnen (DusterTheFirst)
 */

import React, { memo } from "react";
import { Text, View } from "react-native";
import { classComponentStyles } from "../../styles/layout/default";
import { getDisplayColorForBlock } from "../../util/blocks/blockColor";
import { ITimes } from "../../util/class/extentions";
import { IColored } from "../../util/class/primitives";
import usePreferences from "../../util/hooks/usePreferences";

/** A component to display a free block */
function FreeComponent({ start, end, block }: ITimes & IColored) {
    const preferences = usePreferences();

    return (
        <View style={classComponentStyles.container}>
            <View style={classComponentStyles.multiView}>
                <Text style={[classComponentStyles.title, { color: getDisplayColorForBlock(block) }]}>Free</Text>
                <Text style={classComponentStyles.dim}>{start.format("h:mm")} - {end.format("h:mm A")}</Text>
            </View>
            <View style={[classComponentStyles.multiView, classComponentStyles.info]}>
                {preferences.accessibility.labelColors ? <Text style={[classComponentStyles.dim, classComponentStyles.multiMiddle, classComponentStyles.colorblindColor, preferences.accessibility.matchLabelColors ? { color: getDisplayColorForBlock(block) } : undefined]}>{block}</Text> : null}
            </View>
        </View>
    );
}

export default memo(FreeComponent);