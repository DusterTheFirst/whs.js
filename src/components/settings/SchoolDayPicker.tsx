/*!
 * Copyright (C) 2018-2020  Zachary Kohnen (DusterTheFirst)
 */

import { BlockColor } from "@whsha/classes/v1/blocks/blockColor";
import { SchoolDay } from "@whsha/classes/v1/calendar/types";
import { IrregularMeetDays } from "@whsha/classes/v1/class/primitives";
import { getSchoolDaysThatHaveColor } from "@whsha/classes/v1/schoolDays";
import React, { memo, useEffect, useMemo, } from "react";
import { Section } from "react-native-tableview-simple";
import { ButtonCell } from "../../styles/components/tableview";

/** The props for a SchoolDayPicker */
interface ISchoolDayPickerProps {
    /** The callback for when a day is toggled */
    onToggle(day: SchoolDay): void;
    /** The value of the picker */
    value: IrregularMeetDays;
    /** The restraint to lock the block colors to */
    blockColorRestraint: BlockColor;
}

/** A component for picking the school days that a class meets on */
export default function SchoolDayPicker({ onToggle, value, blockColorRestraint }: ISchoolDayPickerProps) {
    const schoolDays = useMemo(() => getSchoolDaysThatHaveColor(blockColorRestraint), [blockColorRestraint]);

    return (
        <Section header="School Days">
            <DayPicker day={SchoolDay.One} value={value} toggle={onToggle} enabledDays={schoolDays} />
            <DayPicker day={SchoolDay.Two} value={value} toggle={onToggle} enabledDays={schoolDays} />
            <DayPicker day={SchoolDay.Three} value={value} toggle={onToggle} enabledDays={schoolDays} />
            <DayPicker day={SchoolDay.Four} value={value} toggle={onToggle} enabledDays={schoolDays} />
            <DayPicker day={SchoolDay.Five} value={value} toggle={onToggle} enabledDays={schoolDays} />
            <DayPicker day={SchoolDay.Six} value={value} toggle={onToggle} enabledDays={schoolDays} />
            <DayPicker day={SchoolDay.Seven} value={value} toggle={onToggle} enabledDays={schoolDays} />
        </Section>
    );
}

/** The props for DayPicker */
interface IDayPickerProps<D extends SchoolDay> {
    /** The day for the picker */
    day: D;
    /** The current selected value */
    value: IrregularMeetDays;
    /** The callback to use to pick a new day */
    toggle(day: D): void;
    /** The school days that are enabled */
    enabledDays: SchoolDay[];
}

/** The cell to choose colors with */
const DayPicker = memo(<D extends SchoolDay>({ day, enabledDays, toggle, value }: IDayPickerProps<D>) => {
    const disabled = !enabledDays.includes(day);
    const selected = value[day] === true;
    const toggleCallback = () => toggle(day);

    useEffect(() => {
        if (disabled && selected) {
            toggleCallback();
        }
    }, [disabled, selected]);

    return (
        <ButtonCell
            title={`Day ${day}`}
            onPress={toggleCallback}
            accessory={selected ? "Checkmark" : undefined}
            isDisabled={disabled}
        />
    );
});