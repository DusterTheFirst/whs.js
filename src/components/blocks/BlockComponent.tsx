/*!
 * Copyright (C) 2018-2020  Zachary Kohnen (DusterTheFirst)
 */

import React, { memo } from "react";
import { IDR, IMajor, IMinor } from "../../util/class/classes";
import { ITimes } from "../../util/class/extentions";
import { IColored } from "../../util/class/primitives";
import { isMajor, isMinor } from "../../util/class/type";
import ClassComponent from "./ClassComponent";
import FreeComponent from "./FreeComponent";

/** The parameters to pass to a BlockComponent */
interface IDisplayBlock extends IColored, ITimes {
    /** The class to display */
    clazz?: IMajor | IMinor | IDR;
}

/** A component that will display a block of many variety */
function BlockComponent({ end, start, clazz, block }: IDisplayBlock) {
    if (clazz === undefined) {
        return <FreeComponent start={start} end={end} block={block} />;
    } else {
        if (isMajor(clazz) || isMinor(clazz)) {
            return (
                <ClassComponent
                    start={start}
                    end={end}
                    name={clazz.name}
                    block={clazz.block}
                    room={clazz.room}
                    teacher={clazz.teacher}
                />
            );
        } else {
            return (
                <ClassComponent
                    start={start}
                    end={end}
                    name="Directed Reaserch"
                    block={clazz.block}
                    room={clazz.room}
                    teacher={clazz.teacher}
                />
            );
        }
    }
}

export default memo(BlockComponent);