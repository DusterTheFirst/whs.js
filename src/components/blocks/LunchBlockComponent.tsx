/*!
 * Copyright (C) 2018-2020  Zachary Kohnen (DusterTheFirst)
 */

import { BlockColor } from "@whsha/classes/v2/block";
import { IClass } from "@whsha/classes/v2/class";
import { Lunch } from "@whsha/classes/v2/lunch";
import dayjs from "dayjs";
import { default as useCustomFormat } from "dayjs/plugin/customParseFormat";
import React, { memo } from "react";
import BlockComponent from "./BlockComponent";
import LunchComponent from "./LunchComponent";

dayjs.extend(useCustomFormat);

/** The parameters to pass to a LunchBlockComponent */
interface ILunchBlock {
    /** The class to display */
    lunch: Lunch;
    /** The class to display */
    clazz?: IClass;
    /** The block color */
    block: BlockColor;
    /** The timings for the classes */
    // times: {
    //     /** The first part of the lunch block */
    //     A: ITimes;
    //     /** The second part of the lunch block */
    //     B: ITimes;
    //     /** The third part of the lunch block */
    //     C: ITimes;
    // };
}

// FIXME: Correct timings
/** The timings for the lunch sections */
const LUNCH_TIMES = {
    A: {
        end: dayjs("11:22 AM", "h:mm A"),
        start: dayjs("10:55 AM", "h:mm A")
    },
    B: {
        end: dayjs("11:52 AM", "h:mm A"),
        start: dayjs("11:25 AM", "h:mm A")
    },
    C: {
        end: dayjs("12:22 PM", "h:mm A"),
        start: dayjs("11:55 AM", "h:mm A")
    }
};

/** A component that will display a block of many variety */
function LunchBlockComponent({ lunch, clazz, block }: ILunchBlock) {
    if (lunch === Lunch.First) {
        return (
            <>
                <LunchComponent start={LUNCH_TIMES.A.start} end={LUNCH_TIMES.A.end} />
                <BlockComponent start={LUNCH_TIMES.B.start} end={LUNCH_TIMES.C.end} block={block} clazz={clazz} />
            </>
        );
    } else if (lunch === Lunch.Second) {
        return (
            <>
                <BlockComponent start={LUNCH_TIMES.A.start} end={LUNCH_TIMES.A.end} block={block} clazz={clazz} />
                <LunchComponent start={LUNCH_TIMES.B.start} end={LUNCH_TIMES.B.end} />
                <BlockComponent start={LUNCH_TIMES.C.start} end={LUNCH_TIMES.C.end} block={block} clazz={clazz} />
            </>
        );
    } else if (lunch === Lunch.Third) {
        return (
            <>
                <BlockComponent start={LUNCH_TIMES.A.start} end={LUNCH_TIMES.B.end} block={block} clazz={clazz} />
                <LunchComponent start={LUNCH_TIMES.C.start} end={LUNCH_TIMES.C.end} />
            </>
        );
    } else {
        return (
            <BlockComponent start={LUNCH_TIMES.A.start} end={LUNCH_TIMES.C.end} block={block} clazz={clazz} />
        );
    }
}

export default memo(LunchBlockComponent);