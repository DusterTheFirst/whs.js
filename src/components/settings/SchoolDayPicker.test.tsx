/*!
 * Copyright (C) 2018-2019  Zachary Kohnen (DusterTheFirst)
 */

import React from "react";
import { render } from "react-native-testing-library";
import { BlockColor } from "../../util/blocks/blockColor";
import { SchoolDay } from "../../util/calendar/types";
import SchoolDayPicker from "./SchoolDayPicker";

function meets(one: boolean, two: boolean, three: boolean, four: boolean, five: boolean, six: boolean, seven: boolean) {
    return {
        [SchoolDay.One]: one,
        [SchoolDay.Two]: two,
        [SchoolDay.Three]: three,
        [SchoolDay.Four]: four,
        [SchoolDay.Five]: five,
        [SchoolDay.Six]: six,
        [SchoolDay.Seven]: seven
    };
}

describe("Tests different variations of <SchoolDayPicker>", () => {
    it("Renders SchoolDayPicker", () => {
        const callback = jest.fn();
        const comp = render(<SchoolDayPicker value={meets(true, true, false, false, false, true, true)} onToggle={callback} blockColorRestraint={BlockColor.None} />);

        expect(comp.toJSON()).toMatchSnapshot();
        expect(callback).toHaveBeenCalledTimes(0);
    });

    it("Renders SchoolDayPicker with a restaint and corrects value", () => {
        const callback = jest.fn();
        const comp = render(<SchoolDayPicker value={meets(true, true, false, true, true, true, true)} onToggle={callback} blockColorRestraint={BlockColor.Green} />);

        expect(comp.toJSON()).toMatchSnapshot();
        expect(callback).toHaveBeenCalledTimes(2);
    });
});