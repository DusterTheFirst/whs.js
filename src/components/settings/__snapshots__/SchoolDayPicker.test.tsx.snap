// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests different variations of <SchoolDayPicker> Renders SchoolDayPicker 1`] = `
<RCTScrollView
  collapsable={false}
  horizontal={true}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "backgroundColor": "white",
              "borderColor": "#CECECE",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "padding": 10,
            },
            Object {
              "backgroundColor": "black",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "height": 20,
                "textAlign": "center",
                "width": 20,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          1
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "backgroundColor": "white",
              "borderColor": "#CECECE",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "padding": 10,
            },
            Object {
              "backgroundColor": "black",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "height": 20,
                "textAlign": "center",
                "width": 20,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          2
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "backgroundColor": "white",
              "borderColor": "#CECECE",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "padding": 10,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "height": 20,
                "textAlign": "center",
                "width": 20,
              },
              undefined,
            ]
          }
        >
          3
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "backgroundColor": "white",
              "borderColor": "#CECECE",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "padding": 10,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "height": 20,
                "textAlign": "center",
                "width": 20,
              },
              undefined,
            ]
          }
        >
          4
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "backgroundColor": "white",
              "borderColor": "#CECECE",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "padding": 10,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "height": 20,
                "textAlign": "center",
                "width": 20,
              },
              undefined,
            ]
          }
        >
          5
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "backgroundColor": "white",
              "borderColor": "#CECECE",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "padding": 10,
            },
            Object {
              "backgroundColor": "black",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "height": 20,
                "textAlign": "center",
                "width": 20,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          6
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "backgroundColor": "white",
              "borderColor": "#CECECE",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "padding": 10,
            },
            Object {
              "backgroundColor": "black",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "height": 20,
                "textAlign": "center",
                "width": 20,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          7
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Tests different variations of <SchoolDayPicker> Renders SchoolDayPicker with a restaint and corrects value 1`] = `
<RCTScrollView
  collapsable={false}
  horizontal={true}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "backgroundColor": "white",
              "borderColor": "#CECECE",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "padding": 10,
            },
            Object {
              "backgroundColor": "black",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "height": 20,
                "textAlign": "center",
                "width": 20,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          1
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "backgroundColor": "white",
              "borderColor": "#CECECE",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "padding": 10,
            },
            Object {
              "backgroundColor": "black",
            },
            Object {
              "backgroundColor": "grey",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "height": 20,
                "textAlign": "center",
                "width": 20,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          2
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "backgroundColor": "white",
              "borderColor": "#CECECE",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "padding": 10,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "height": 20,
                "textAlign": "center",
                "width": 20,
              },
              undefined,
            ]
          }
        >
          3
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "backgroundColor": "white",
              "borderColor": "#CECECE",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "padding": 10,
            },
            Object {
              "backgroundColor": "black",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "height": 20,
                "textAlign": "center",
                "width": 20,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          4
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "backgroundColor": "white",
              "borderColor": "#CECECE",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "padding": 10,
            },
            Object {
              "backgroundColor": "black",
            },
            Object {
              "backgroundColor": "grey",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "height": 20,
                "textAlign": "center",
                "width": 20,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          5
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "backgroundColor": "white",
              "borderColor": "#CECECE",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "padding": 10,
            },
            Object {
              "backgroundColor": "black",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "height": 20,
                "textAlign": "center",
                "width": 20,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          6
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "backgroundColor": "white",
              "borderColor": "#CECECE",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "padding": 10,
            },
            Object {
              "backgroundColor": "black",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "height": 20,
                "textAlign": "center",
                "width": 20,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          7
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
